keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1180 @@
+#### Trust Anchor Configuration for Aggregated API Server Backends
+
+The `apiregistration.k8s.io` v1 APIServiceSpec object will be extended with a
+new optional field `TrustAnchors` that references a single trust anchor set
+within a single ClusterTrustBundle object.  `TrustAnchors` will be mutually
+exclusive with the existing `CABundle` and `InsecureSkipTLSVerify` fields.
+
+When `TrustAnchors` is specified, kube-apiserver will load the certificates from
+the named trust anchor set of the named ClusterTrustBundle object, an use those
+for validating the certificate presented by the aggregated API server.
+kube-apiserver will follow updates to the ClusterTrustBundle object, and use the
+updated certificates within a few seconds.
+
+```diff
+type APIServiceSpec struct {
+    // ...
+
+    // InsecureSkipTLSVerify disables TLS certificate verification when communicating with this server.
+    // This is strongly discouraged.  You should use the CABundle instead.
+    InsecureSkipTLSVerify bool `json:"insecureSkipTLSVerify,omitempty" protobuf:"varint,4,opt,name=insecureSkipTLSVerify"`
+    // CABundle is a PEM encoded CA bundle which will be used to validate an API server's serving certificate.
+    // If unspecified, system trust roots on the apiserver are used.
+    // +listType=atomic
+    // +optional
+    CABundle []byte `json:"caBundle,omitempty" protobuf:"bytes,5,opt,name=caBundle"`
++	// A reference to a ClusterTrustBundle that will be used to validate
++	// the webhook's server certificate.
++	//
++	// Mutually exclusive with CABundle and InsecureSkipTLSVerify.
++	//
++	// This field is alpha.  Using it requires setting the ClusterTrustBundle
++	// feature gate.
++	//
++	// +optional
++	ClusterTrustBundleName string `json:"clusterTrustBundleName,omitempty"`
+
+    // ...
+}
+```
+
+### pemTrustAnchors Projected Volume Source
+
+The pemTrustAnchors projected volume source is responsible for writing the
+contents of a particular ClusterTrustBundle object into the container filesystem
+as a single file containing a verbatim copy of the correct
+`.spec.pemTrustAnchors` field.
+
+All of the new fields described in this section are gated by the
+ClusterTrustBundle kubelet feature gate.  Unless the feature gate is enabled,
+usage of these alpha objects and fields will be rejected by kubelet.
+
+#### Configuration Object Definition
+
+```go
+// PEMTrustAnchorsProjection is the top-level configuration object for
+// instances of the trustAnchors kubelet projected volume source.
+type PEMTrustAnchorsProjection struct {
+  // An explicit ClusterTrustBundle to retrieve trust anchor sets from.
+  ClusterTrustBundleName string `json:"clusterTrustBundleName"`
+
+  // Where should we write the flat file in the projected volume?
+  Path string `json:"path"`
+}
+```
+
+#### Volume Content Generation and Refresh
+
+For each pemTrustAnchors projected volume source in the pod spec:
+
+1. Kubelet will establish a watch on the named ClusterTrustBundle (using a
```

Comment:
I've simplified this section so that it doesn't imply anything about *how* kubelet actually gets the updates.  The amount is probably low enough that we can just have a global watch.  The objects will be fairly big, though.